src/node_sea.cc

```diff
@@ -271,6 +271,9 @@ std::tuple<int, char**> FixupArgsForSEA(int argc, char** argv) {
     new_argv.emplace_back(argv[0]);
     new_argv.insert(new_argv.end(), argv, argv + argc);
     new_argv.emplace_back(nullptr);
+#ifdef _WIN32
+    delete[] argv;
```

Comment:
`FixupArgsForSEA` is not the right place to handle it, this branch is only hit in an SEA and isn't hit in usual paths - also it seems the inner arrays are still not deleted properly. I think we could either delete them all in wmain after `node::Start()` returns, or just turn them into `std::unique_ptr<std::unique_ptr<char[]>[]>` in `wmain` to make them a bit more automatic.

Reply:
Actually libuv might take ownership of the argv (it's hard to tell exactly what would happen when the libuv docs only say "Libuv *may* take ownership of the memory that argv points to", we are doing `new []` but libuv definitely has not been doing `delete[]` on them). I think a safer approach is probably moving the `uv_setup_args` call in `StartInternal()` to `Start()`, and in the `DISABLE_SINGLE_EXECUTABLE_APPLICATION` branch, delete the original argv if the one returned by `FixupArgsForSEA` does not equal to the one being returned under Windows.

Reply:
The inner data of argv is been set into new_argv, so we should not delete the content of argv

Reply:
If libuv doesn't `delete []` the array content (it is unlikely to use delete at all since it's in C, in reality on Windows [`uv_setup_args`](https://github.com/nodejs/node/blob/36dcd399c05fe4f25c759e5c54aac9c10d2b98f8/deps/uv/src/win/util.c#L342) just seems to be a noop), won't the content still be leaked after all because nobody will delete it ever?

Also note that `node::Start()` is not only used by the Node.js executable, it is also exposed in the embedder API. There hasn't really been any documentation about what Node.js would do to the argv...

Reply:
> Also note that node::Start() is not only used by the Node.js executable, it is also exposed in the embedder API. There hasn't really been any documentation about what Node.js would do to the argv...

Ah! I see